casper/src/main/scala/coop/rchain/casper/engine/LastFinalizedStateBlockRequester.scala

```diff
@@ -72,24 +89,88 @@ object LastFinalizedStateBlockRequester {
     * @param responseQueue Handler of block messages
     * @return fs2.Stream processing all blocks
     */
-  def stream[F[_]: Concurrent: Time: BlockDagStorage: BlockStore: CommUtil: Log](
+  def stream[F[_]: Concurrent: Time: BlockStore: CommUtil: Log](
       approvedBlock: ApprovedBlock,
       responseQueue: Queue[F, BlockMessage]
   ): F[Stream[F, Boolean]] = {
 
+    val block                           = approvedBlock.candidate.block
+    val approvedBlockNumber             = ProtoUtil.blockNumber(block)
+    val minBlockNumberForDeployLifespan = approvedBlockNumber - MultiParentCasperImpl.deployLifespanBlocks
```

Comment:
This safety range should have some name. Maybe DSR? deploy safety range? so this might be dsrBottomHeight? Im ok leaving this as it is, just think we need a name for this 50 blocks t cont are actually configurable